File: src/main/java/sn/brasilburger/Entity/CommandeItem.java
package sn.brasilburger.Entity;

public class CommandeItem {
    private int id;
    private int commandeId;
    private int burgerId;
    private int menuId;
    private int complementId;
    private int quantite;
    private Double prixTotal;

    public CommandeItem() {
    }

    public CommandeItem(int id, int commandeId, int burgerId, int menuId, int complementId, int quantite, Double prixTotal) {
        this.id = id;
        this.commandeId = commandeId;
        this.burgerId = burgerId;
        this.menuId = menuId;
        this.complementId = complementId;
        this.quantite = quantite;
        this.prixTotal = prixTotal;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCommandeId() {
        return commandeId;
    }

    public void setCommandeId(int commandeId) {
        this.commandeId = commandeId;
    }

    public int getBurgerId() {
        return burgerId;
    }

    public void setBurgerId(int burgerId) {
        this.burgerId = burgerId;
    }

    public int getMenuId() {
        return menuId;
    }

    public void setMenuId(int menuId) {
        this.menuId = menuId;
    }

    public int getComplementId() {
        return complementId;
    }

    public void setComplementId(int complementId) {
        this.complementId = complementId;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public Double getPrixTotal() {
        return prixTotal;
    }

    public void setPrixTotal(Double prixTotal) {
        this.prixTotal = prixTotal;
    }
}

File: CSharp/Models/CommandeItem.cs
namespace Models
{
    using System.ComponentModel.DataAnnotations.Schema;
    public class CommandeItem
    {
        [Column("id")]
        public long Id { get; set; }
        [Column("commande_id")]
        public long CommandeId { get; set; }
        [Column("burger_id")]
        public long? BurgerId { get; set; }
        [Column("menu_id")]
        public long? MenuId { get; set; }
        [Column("complement_id")]
        public long? ComplementId { get; set; }
        [Column("quantite")]
        public int Quantite { get; set; }
        [Column("prix_total")]
        public double PrixTotal { get; set; }

        [ForeignKey("BurgerId")]
        public Burger? Burger { get; set; }

        [ForeignKey("MenuId")]
        public Menu? Menu { get; set; }

        [ForeignKey("ComplementId")]
        public Complement? Complement { get; set; }
    }

}